api/src/main/java/jakarta/data/repository/Is.java

```diff
@@ -0,0 +1,120 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.repository;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.data.constraint.AtLeast;
+import jakarta.data.constraint.AtMost;
+import jakarta.data.constraint.Constraint;
+import jakarta.data.constraint.EqualTo;
+import jakarta.data.constraint.GreaterThan;
+import jakarta.data.constraint.In;
+import jakarta.data.constraint.LessThan;
+import jakarta.data.constraint.Like;
+import jakarta.data.constraint.NotEqualTo;
+import jakarta.data.constraint.NotIn;
+import jakarta.data.constraint.NotLike;
+
+/**
+ * <p>Annotates a parameter of a repository {@link Find} or {@link Delete}
+ * method, indicating how an entity attribute is compared with the parameter's
+ * value.</p>
+ *
+ * <p>The {@code @Is} annotation's {@link #value()} supplies the type of
+ * comparison as a subtype of {@link jakarta.data.constraint Constraint}.</p>
+ *
+ * <p>The {@link By} annotation must annotate the same parameter to indicate
+ * the entity attribute name, or otherwise, if the {@code -parameters} compile
+ * option is enabled, the persistent field is inferred by matching the name of
+ * the method parameter.</p>
+ *
+ * <p>For example,</p>
+ *
+ * <pre>
+ * &#64;Repository
+ * public interface Products extends CrudRepository&lt;Product, Long&gt; {
+ *
+ *     // Find Product entities where the price attribute is less than a maximum value.
+ *     &#64;Find
+ *     List&lt;Product&gt; pricedBelow(&#64;By(_Product.PRICE) &#64;Is(LessThan.class) float max);
+ *
+ *     // Find a page of Product entities where the name field matches a pattern.
+ *     &#64;Find
+ *     Page&lt;Product&gt; search(&#64;By(_Product.NAME) &#64;Is(Like.class) String pattern,
+ *                          PageRequest pagination,
+ *                          Order&lt;Product&gt; order);
+ *
+ *     // Remove Product entities with any of the unique identifiers listed.
+ *     &#64;Delete
+ *     void remove(&#64;By(ID) &#64;Is(In.class) List&lt;Long&gt; productIds);
+ * }
+ * </pre>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface Is {
+
+    /**
+     * <p>A subtype of {@link jakarta.data.constraint Constraint} that
+     * indicates how the entity attribute is compared with a value.</p>
+     *
+     * <p>The constraint subtype must have a static builder method that accepts
```

Comment:
I would stay away from the undefined term "builder method". Just saying: "a static method" is enough, I think.